msg.go

```diff
@@ -71,6 +71,17 @@ var (
 	ErrTime          error = &Error{err: "bad time"}      // ErrTime indicates a timing error in TSIG authentication.
 )
 
+type OverflowError struct {
+	err string
```

Comment:
I think this should be *Error and then OverflowError should implement the Unwrap method. Otherwise all of the errors you changed will be a breaking change since they won't be *Error anymore.

Reply:
@jameshartig - I have implemented the changes. Could you please take another look ?

Reply:
@tmthrgd - Would you mind reviewing this PR please ?

Reply:
As far as I know, I have addressed all feedback and this is ready for review.

Reply:
Hi Team, I kindly request a review on this PR. This is blocking another PR for coreDNS.

Reply:
@tmthrgd - I kindly request you to review this PR.

Reply:
@jameshartig @tmthrgd - I apologize for multiple follow up. I kindly request you to review this PR when you get a moment.

Reply:
Hi Team, please review this PR when you get a chance.